app_dart/lib/src/request_handlers/github/webhook_subscription.dart

```diff
@@ -840,3 +870,122 @@ class GithubWebhookSubscription extends SubscriptionHandler {
     return true;
   }
 }
+
+typedef PullRequestLabelProcessorProvider = PullRequestLabelProcessor Function({
+  required Config config,
+  required GithubService githubService,
+  required PullRequest pullRequest,
+});
+
+class PullRequestLabelProcessor {
+  PullRequestLabelProcessor({
+    required this.config,
+    required this.githubService,
+    required this.pullRequest,
+  }) : slug = pullRequest.base!.repo!.slug();
+
+  final Config config;
+  final GithubService githubService;
+  final PullRequest pullRequest;
+  final RepositorySlug slug;
+
+  int get prNumber => pullRequest.number!;
+
+  String get logCrumb => 'PullRequestLabelProcessor($slug/pull/$prNumber)';
+
+  void logInfo(Object? message) {
+    log.info('$logCrumb: $message');
+  }
+
+  void logSevere(Object? message) {
+    log.severe('$logCrumb: $message');
+  }
+
+  Future<void> processLabels() async {
+    final hasEmergencyLabel = pullRequest.labels?.any((label) => label.name == Config.kEmergencyLabel) ?? false;
+    if (hasEmergencyLabel) {
+      await processEmergencyLabel();
+    } else {
+      logInfo('no emergency label; moving on.');
+    }
+  }
+
+  Future<void> processEmergencyLabel() async {
+    final hasApprovals = await computeTeamMemberApproval();
+
+    if (!hasApprovals) {
+      // Do not apply emergency label without an approval from a team member.
+      logInfo('No team member approval yet.');
```

Comment:
It would be nice if we removed the label and added a message why we removed it.

I understand that might not be feasible though, but I can definitely see this causing "I added the label but don't see it working @ YEGOR HELP.